packages/react-app/pages/index.jsx

```diff
@@ -64,6 +68,7 @@ export default function Index({ bgStats, events }) {
 export async function getStaticProps() {
   const stats = await getStats();
   const events = await getAllEvents(null, 10);
+  const builds = (await getAllBuilds()).sort((a, b) => b.submittedTimestamp - a.submittedTimestamp).slice(0, 4);
```

Comment:
Lol did a bit cheating here(doing all calculation's here)

 Should we update `getAllBuilds` such that it takes `limit` as second arg, we will need to propagate that `limit` till backend API and use firestore `limit` query along with `orderBy` (so that we get the latest submittedAt) 
 
 But since this will only run after approx every 6hr it's fine ? 
 
 We are also planning to decrease `revalidate` , so if we plan to keep it around 15min / 30min maybe we optimize this query by passing `limit` and update the API, so that we don't get charged for reading all builds read approx every 15 / 30min ?  
 
 Lol but not sure what's the best, but happy to update the API incase we plan to ! 🙌 
 
 ^ I think we will need to index `submittedTimestamp` from firestore UI if we plan to go that route

Reply:
I think it's ok for now!

> We are also planning to decrease revalidate , so if we plan to keep it around 15min / 30min maybe we optimize this

I added a comment on that PR too. I think we should do 2/3 hours... or just keep at 6 and fetch the activity in the frontend

Reply:
> I added a comment on that PR too. I think we should do 2/3 hours... or just keep at 6 and fetch the activity in the frontend

Just say it, yup makes sense 🙌 !